docs/reference/search/term-enum.asciidoc

```diff
@@ -0,0 +1,101 @@
+[[search-term-enum]]
+=== Term enum API
+
+The term enum API can be used to discover terms in the index that match
+a partial string. This is used for features like auto-complete:
+
+[source,console]
+--------------------------------------------------
+POST stackoverflow/_terms
+{
+    "field" : "tags",
+    "string" : "kiba"
+}
+--------------------------------------------------
+// TEST[setup:stackoverflow]
+
+
+The API returns the following response:
+
+[source,console-result]
+--------------------------------------------------
+{
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "failed": 0
+  },
+  "terms": [
+    "kibana"
+  ],
+  "timed_out" : false
+}
+--------------------------------------------------
+// TESTRESPONSE[s/8/$body.terms.0.doc_count/]
+
+[[search-term-enum-api-request]]
+==== {api-request-title}
+
+`GET /<target>/_terms`
+
+
+[[search-term-enum-api-desc]]
+==== {api-description-title}
+
+The termenum API  can be used to discover terms in the index that begin with the provided
+string. It is designed for low-latency look-ups used in auto-complete scenarios.
+
+
+[[search-term-enum-api-path-params]]
+==== {api-path-parms-title}
+
+`<target>`::
+(Mandatory, string)
+Comma-separated list of data streams, indices, and index aliases to search.
+Wildcard (`*`) expressions are supported.
++
+To search all data streams or indices in a cluster, omit this parameter or use
+`_all` or `*`.
```

Comment:
Should we forbid searching all data ? Plain `*` or omitting the pattern could return an error ?

Reply:
We have the tier filtering now so this so searching all may be more viable